components/server/src/session-handler.ts

```diff
@@ -43,11 +43,25 @@ export class SessionHandlerProvider {
     }
 
     protected getCookieOptions(env: Env): express.CookieOptions {
-        const hostParts = env.hostUrl.url.host.split('.');
-        const baseDomain = hostParts.slice(hostParts.length - 2).join('.');
-        let domain = `.${baseDomain}`;
+        const hostName = env.hostUrl.url.host;
+
+        let domain = hostName;
+        if (env.devBranch) {
+            // Use cookie for base domain to allow cookies being sent via ingress proxy in preview environments
+            //
+            // Otherwise, clients (in this case Chrome) may ignore (as in: save it, but don't send it on consequent requests) the 'Set-Cookie:...' send with a redirect (302, to github oauth)
+            // For details, see:
+            // - RFC draft sameSite: http://httpwg.org/http-extensions/draft-ietf-httpbis-cookie-same-site.html
+            // - https://bugs.chromium.org/p/chromium/issues/detail?id=150066
+            // - google: chromium not sending cookies set with redirect
+            
+            const hostParts = hostName.split('.');
+            const baseDomain = hostParts.slice(hostParts.length - 2).join('.');
```

Comment:
I wonder why the slicing and special handling is necessary here. Also in dev staging all the requests should go to the hostname URL. Is that not the case?

Reply:
exactly. I recall, why this is actually needed. in dev staging preview environments we have the ingress proxy which terminates https connections, which is a crucial part in the login process, right? the URL of the ingress proxy has no namespace prefix at all, that's why we've chosen the sliced version of it. this allows those cookies to be present on  ingress proxy requests, which are then forwarded to the services.